src/Framework/Framework/Routing/RouteBase.cs

```diff
@@ -203,11 +194,13 @@ public static void AddOrUpdateParameterCollection(IDictionary<string, object?> t
             }
         }
 
-
         /// <summary>
         /// Returns a presenter that processes the request.
         /// </summary>
-        public abstract IDotvvmPresenter GetPresenter(IServiceProvider provider);
+        public virtual IDotvvmPresenter GetPresenter(IServiceProvider provider) => PresenterFactory(provider);
+
+        public Dictionary<string, object?> CloneDefaultValues() => new Dictionary<string, object?>(_defaultValues, StringComparer.OrdinalIgnoreCase);
```

Comment:
[nitpick] Could be worth it to push the cloning to FreezableDictionary, then we could use Dictionary.Comparer to make sure we use the same one as the original dict, and calling new Dictionary(dict) is faster if `dict.GetType()==typeof(System.Collection.Generic.Dictionary<...>)`